### ISSUE TITLE
Allow to limit the reporting of deprecations, notices, and warnings to specified directories

### ISSUE THREAD

**Issue by sebastianbergmann** [sebastianbergmann/phpunit#5293]
As of PHPUnit 10.0, the test runner prints `D`, `N`, and `W`, respectively, for tests that execute code which triggers `E_DEPRECATED`, `E_USER_DEPRECATED`, `E_NOTICE`, `E_USER_NOTICE`, `E_STRICT`, `E_WARNING`, or `E_USER_WARNING` events.

Detailed information (which notices was triggered where, for instance) is only printed when `--display-deprecations`, `--display-notices`, or `--display-warnings` is used.

The test runner should support limiting the reporting of deprecations, notices, and warnings to "your code", excluding code from directories such as `vendor`, for example.

This depends on #5294.

**Comment by acelaya** [comment]
Commenting here as I was notified by this superseding another issue I opened some time ago: https://github.com/sebastianbergmann/phpunit/issues/5191

> The test runner should support limiting the reporting of deprecations, notices, and warnings to "your code", excluding code from directories such as vendor, for example.

I actually think it makes more sense the other way around. It can be useful to be notified about the usage of deprecated third party code, as it's easy for the deprecations to be unknown until it's "too" late.

As per the project's own code, the usage of deprecated symbols is probably known and there's plans to stop using them. It is also likely they were deprecated just to avoid using them by mistake in new code, but you don't need to be warned about your own code.

But of course I'm biased here and probably missing the big picture 😅
